logstash-core/lib/logstash/api/lib/app/service.rb

```diff
@@ -40,4 +44,20 @@ def get(key)
     end
     LogStash::Json.dump(data)
   end
+
+  private
+
+  def has_counters?
+    return false if @snapshot.nil?
+    if @snapshot_rotation_mutex.try_lock
```

Comment:
the mutex is locked we return false here, is that what we want? Why not block until it's free?

```ruby
@snapshot_rotation_mutex.synchronize do
  metric_types.include?("LogStash::Instrument::MetricType::Counter") &&
  metric_types.include?("LogStash::Instrument::MetricType::Gauge")
end
```